pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -619,4 +645,52 @@ public void close()
     creators.addAll(_dictionaryCreatorMap.values());
     FileUtils.close(creators);
   }
+
+  /**
+   * Returns the original string if its length is within the allowed limit.
+   * If the string's length exceeds the limit,
+   * it returns a truncated version of the string with maintaining min or max value.
+   *
+   */
+  @VisibleForTesting
+  static String getValueWithinLengthLimit(String value, boolean isMax, DataType dataType) {
+    int length = value.length();
+
+    // if length is less, no need of trimming the value.
+    if (length < METADATA_PROPERTY_LENGTH_LIMIT) {
+      return value;
+    }
+
+    String alteredValue;
+    // For Numeric Data Type(INT, LONG, DOUBLE, FLOAT) value longer than METADATA_PROPERTY_LENGTH_LIMIT is not possible.
+    switch (dataType) {
+      case STRING:
+      case JSON:
+        if (isMax) {
+          alteredValue = value.substring(0, METADATA_PROPERTY_LENGTH_LIMIT - 1)
+              + (char) (value.charAt(METADATA_PROPERTY_LENGTH_LIMIT - 1) + 1);
```

Comment:
I think it's not possible; the last char, as per Ascii, is `DEL`(127), and I think that wouldn't be part of any string value.

Reply:
This is java `char`, which is 16 bytes and contains a lot of special characters. I don't know if all of the chars are printable and can be stored in the `Configuration`. We need to avoid creating unprintable char here

Reply:
Instead of incrementing char, I have updated the code to add `z` as a last character. Please let me know if it works or suggest an alternative please.  Thanks

Reply:
`z` is not good enough because it is actually a very small value (122) comparing to the 16-bit unicode space. I tried appending '\uFFFF' and seems `PropertiesConfiguration` can store and retrieve it properly. We can just loop over the characters, and replace the first one that is not '\uFFFF' to '\uFFFF'